Enable touch driver tma525b

KATE-WANG-NXP opened the pull request:
1.Add support for the Parade Tech TMA525B capacitive touch controller. The driver supports both interrupt-driven and polling modes, and can handle up to 4 simultaneous touch points.
2.Add support for the TMA525B touch controller to the  zc143ac72mipi display shield.
3.Adds the zc143ac72mipi shield configuration for the mimxrt700_evk platform to the LVGL demos sample test platforms

KATE-WANG-NXP committed:
samples: lvgl: demos: Add zc143ac72mipi shield to test platforms

Adds the zc143ac72mipi shield configuration for the mimxrt700_evk
platform to the LVGL demos sample test platforms.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

pdgendt commented on drivers/input/kconfig.tma525b:
```diff
@@ -0,0 +1,36 @@
+	default y
+	depends on DT_HAS_PARADE_TMA525B_ENABLED
+	select I2C
+    select INPUT_TOUCH
```
```suggestion
	select INPUT_TOUCH
```

pdgendt reviewed (commented):
Fix compliance issues, please

fabiobaltieri commented on drivers/input/input_tma525b.c:
```diff
@@ -0,0 +1,436 @@
+	/* Timestamp in 100us units low byte. Not used in current driver model. */
+	uint8_t timestamp_100us_l;
+	/* Timestamp in 100us units high byte. Not used in current driver model. */
+	uint8_t timestamp_100us_h;
```
use a single `uint16_t`? it's unused anyway

fabiobaltieri commented on drivers/input/input_tma525b.c:
```diff
@@ -0,0 +1,436 @@
+
+/* Macros for extracting touch data. */
+#define TMA525B_TOUCH_POINT_GET_ID(event_id)    (event_id & 0x1F)
+#define TMA525B_TOUCH_POINT_GET_EVENT(event_id) (((event_id) & 0x60) >> 5U)
```
drop the parenthesis from around `event_id`? you don't have them in the line above anyway

fabiobaltieri commented on drivers/input/input_tma525b.c:
```diff
@@ -0,0 +1,436 @@
+		gpio_pin_set_dt(&config->rst_gpio, 0);
+	}
+
+	/* Wait 120ms for boot */
```
sorry I know you changed this already, just drop the time constant, leave "Wait for the device to boot", just comment what we can't infer from the code.

even better would be to add a define for 120, call it like `BOOT_DELAY_MS` and then you don't need the comment at all as the code would be clear enough by itself, yeah do that please :-)

JarmouniA commented on boards/shields/zc143ac72mipi/Kconfig.defconfig:
```diff
@@ -0,0 +1,39 @@
+	default y if !MIPI_DSI_MCUX_2L_SWAP16
+
+config LV_Z_VDB_SIZE
+	default 16
```
ditto

JarmouniA commented on boards/shields/zc143ac72mipi/Kconfig.defconfig:
```diff
@@ -0,0 +1,39 @@
+
+if SHIELD_ZC143AC72MIPI
+
+if LVGL
```
```suggestion
```

JarmouniA commented on boards/shields/zc143ac72mipi/Kconfig.defconfig:
```diff
@@ -0,0 +1,39 @@
+	default y
+
+endif # MIPI_DSI_MCUX_2L
+
```
```suggestion

if LVGL

```

KATE-WANG-NXP committed:
boards: shields: zc143ac72mipi: Add touch controller and LVGL config

- Add support for the TMA525B touch controller to the
zc143ac72mipi shield
- Add Kconfig.defconfig with LVGL configuration defaults
- Update board specific configuration

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

KATE-WANG-NXP committed:
boards: shields: g1120b0mipi: Move configs to board-specific files

Move MIPI_DSI_MCUX_2L_SWAP16 and LV_COLOR_16_SWAP configuration from
the shield's Kconfig.defconfig to board-specific configuration files.
This allows for better board-level customization and removes the
conditional logic from the shield defaults.

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

fabiobaltieri commented on drivers/input/input_tma525b.c:
```diff
@@ -0,0 +1,434 @@
+	uint8_t reg_addr = TMA525B_TOUCH_DATA_SUBADDR;
+	int ret;
+	uint8_t i, j;
+	int x, y;
```
declare these two in the cycle where they are used, just trying to narrow the scope, the automatic variable block here is huge, alternatively you could split the function but moving this would help

```c
	for (i = 0; i < touch_count; i++) {
                int x, y;
...
```

JarmouniA replied:
Yes it's in the Linux kernel coding guidelines.

JarmouniA replied:
I failed to find the reference, maybe it doesn't exist, but in my opinion still a good recommendation, less chance of git conflicts, more readable, the downside is the extra lines.

fabiobaltieri commented on drivers/input/input_tma525b.c:
```diff
@@ -0,0 +1,434 @@
+	int x, y;
+	uint8_t touch_id;
+	enum touch_event event;
+	bool touch_found;
```
same, declare it in the loop where it's used

fabiobaltieri commented on drivers/input/input_tma525b.c:
```diff
@@ -0,0 +1,434 @@
+static int tma525b_chip_init(const struct device *dev)
+{
+	const struct tma525b_config *config = dev->config;
+	uint8_t reg_addr = TMA525B_TOUCH_DATA_SUBADDR;
```
same, use the value directly

KATE-WANG-NXP committed:
drivers: input: Add TMA525B capacitive touch controller driver

This commit adds support for the Parade Tech TMA525B capacitive touch
controller. The driver supports both interrupt-driven and polling modes,
and can handle up to 4 simultaneous touch points.

Key features:
- I2C communication interface
- Multi-touch support (up to 4 touch points)
- Interrupt mode with GPIO callback support
- Polling mode with timer
- Power management support with PM notifier
- Reset and power control via GPIO
- Touch event tracking (down, contact, up)

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

JarmouniA commented on boards/shields/zc143ac72mipi/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+
+if LVGL
+
+config LV_Z_VDB_SIZE
```
board-specific, should be moved to board defconfig.

JarmouniA replied:
Also depends on how much memory the board has.

KATE-WANG-NXP committed:
tests: drivers: build_all: input: Add TMA525B to build test

Add the TMA525B touch controller driver to the input build_all test.
This includes:
- Device tree overlay entry for the TMA525B device
- Test configuration enabling interrupt mode

Signed-off-by: Kate Wang <yumeng.wang@nxp.com>
